url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/Console.java

```diff
@@ -0,0 +1,623 @@
+        options.add(new Option(KW_DATA_MAXIMUM, "number", "if exporting to fld without datafile, maximum number of results to export"));
+        options.add(new Option(KW_DECIMALS, "number", "number of decimals to utilize"));
+
+        StringBuilder result = new StringBuilder();
+        result.append("========================================\n");
+        result.append("fuzzylite: a fuzzy logic control library\n");
+        result.append(String.format("version: %s\n", FuzzyLite.LONG_VERSION));
+        result.append(String.format("author: %s\n", FuzzyLite.AUTHOR));
+        result.append("========================================\n");
+        result.append("usage: java -jar jfuzzylite.jar inputfile outputfile\n");
+        result.append("   or: java -jar jfuzzylite.jar ");
+        for (Option option : options) {
+            result.append(String.format("[%s %s] ", option.key, option.value));
+        }
+        result.append("\n\nwhere:\n");
+
+        for (Option option : options) {
+            result.append(String.format("%s %s \t%s.\n",
+                    option.key, option.value, option.description));
+        }
+        result.append("\n");
+        result.append("Visit http://www.fuzzylite.com for more information.");
+        return result.toString();
+    }
+
+    protected static Map<String, String> parse(String[] args) {
+        if (args.length % 2 != 0) {
+            throw new RuntimeException("[option error] incomplete number of parameters [key value]");
+        }
+        Map<String, String> options = new HashMap<>();
+        String key, value;
+        for (int i = 0; i < args.length - 1; i += 2) {
+            key = args[i];
+            value = args[i + 1];
+            options.put(key, value);
+        }
+        if (options.size() == 1) {
+            Map.Entry<String, String> in_out = options.entrySet().iterator().next();
+            if (in_out.getKey().charAt(0) != '-') {
+                options.put(KW_INPUT_FILE, in_out.getKey());
+                options.put(KW_OUTPUT_FILE, in_out.getValue());
+            }
+        } else {
+            Set<String> validOptions = new HashSet<>();
+            validOptions.add(KW_INPUT_FILE);
+            validOptions.add(KW_INPUT_FORMAT);
+            validOptions.add(KW_OUTPUT_FILE);
+            validOptions.add(KW_OUTPUT_FORMAT);
+            validOptions.add(KW_EXAMPLE);
+            validOptions.add(KW_DATA_INPUT);
+            validOptions.add(KW_DATA_MAXIMUM);
+            validOptions.add(KW_DECIMALS);
+            for (String option : options.keySet()) {
+                if (!validOptions.contains(option)) {
+                    throw new RuntimeException(String.format(
+                            "[option error] option <%s> not supported", option));
+                }
+            }
+        }
+        return options;
+    }
+
+    protected static void process(Map<String, String> options) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 26. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.